crates/universal-sierra-compiler-api/src/lib.rs

```diff
@@ -37,49 +36,47 @@ pub struct AssembledCairoProgramWithSerde {
     pub hints: Vec<(usize, Vec<Hint>)>,
 }
 
-pub fn compile_sierra_to_casm(sierra_program: &Program) -> Result<AssembledProgramWithDebugInfo> {
-    let assembled_with_info_raw = compile_sierra(
-        &serde_json::to_value(sierra_program)?,
-        None,
-        &SierraType::Raw,
-    )?;
-    let assembled_with_info: AssembledProgramWithDebugInfo =
-        serde_json::from_str(&assembled_with_info_raw)?;
+pub trait UscOutputTransformer {
+    fn transform(output: String) -> Self;
+}
+
+impl UscOutputTransformer for String {
+    fn transform(output: String) -> Self {
+        output
+    }
+}
 
-    Ok(assembled_with_info)
+impl UscOutputTransformer for AssembledProgramWithDebugInfo {
+    fn transform(output: String) -> Self {
+        serde_json::from_str(&output).expect("Failed to deserialize casm from json string")
+    }
 }
 
-pub fn compile_sierra(
-    sierra_contract_class: &Value,
-    current_dir: Option<&Path>,
+pub fn compile_sierra<T: UscOutputTransformer>(
+    sierra_json: &Value,
     sierra_type: &SierraType,
-) -> Result<String> {
+) -> Result<T> {
     let mut temp_sierra_file = Builder::new().tempfile()?;
-    let _ = temp_sierra_file.write(serde_json::to_vec(sierra_contract_class)?.as_slice())?;
+    let _ = temp_sierra_file.write(serde_json::to_vec(sierra_json)?.as_slice())?;
 
     compile_sierra_at_path(
-        temp_sierra_file.path().to_str().unwrap(),
-        current_dir,
+        Utf8Path::new(&temp_sierra_file.path().to_string_lossy().to_string()),
```

Comment:
```suggestion
        Utf8Path::from_path(temp_sierra_file.path()).unwrap(),
```
Or instead of unwrap, you can convert it to anyhow error and bubble it. Not sure what makes more sense here, probably the latter since we are bubbling all other tmp file related error.